src/disjoint_set.jl

```diff
@@ -57,22 +57,28 @@ function union!(s::IntDisjointSets, x::Integer, y::Integer)
     parents = s.parents
     xroot = find_root_impl!(parents, x)
     yroot = find_root_impl!(parents, y)
-
     if xroot != yroot
-        rks = s.ranks
-        @inbounds xrank = rks[xroot]
-        @inbounds yrank = rks[yroot]
-
-        if xrank < yrank
-            @inbounds parents[xroot] = yroot
-        else
-            @inbounds parents[yroot] = xroot
-            if xrank == yrank
-                rks[xroot] += 1
-            end
-        end
-        @inbounds s.ngroups -= 1
+        _union!(s, xroot, yroot)
+    end
```

Comment:
If `xroot == yroot`, nothing will be returned.  For type-stability, it should return `xroot`.